cmd/algokey/sign.go

```diff
@@ -71,6 +74,9 @@ var signCmd = &cobra.Command{
 			}
 
 			stxn.Sig = key.Sign(stxn.Txn)
+			if rekeyed {
+				stxn.AuthAddr = basics.Address(key.SignatureVerifier)
+			}
```

Comment:
To make sure I understand. You sign the Transaction and then modify the structure? Why don't we sign the rekeyed param?

Reply:
The AuthAddr is part of the SignedTxn but not part of the Transaction -- it doesn't get signed. (The AuthAddr doesn't change what the transaction does, it just tells you what address to check the sig / msig / lsig against.)

Reply:
It sounds like the "AuthAddr" isn't even required here, as the node could see if it's defined in the account. If it does, it could use it against the signature anyway.

Reply:
@tsachiherman If I remember correctly, this is was a performance decision tradeoff.

Reply:
Yes, the AuthAddr is deliberately included in the SignedTxn explicitly because otherwise it makes signature checks stateful, and a lot of the optimizations we have for the block evaluator need signature checking to be stateless. (We went with stateless TEAL for similar reasons.) Explicitly including the AuthAddr does make SignedTxns a little bigger, but only if they're from accounts that have rekeyed, which are expected to be rare. The design doc discusses this in a little more detail.